lua/agentic/session_restore_race.test.lua

```
@@ -289,31 +303,115 @@ describe("race: stale create_session after load_acp_session", function()
+        session:new_session()
+
+        assert.is_true(vim.tbl_contains(session._cancelled, "restored-id"))
+        assert.is_nil(session._restoring_session_id)
+        assert.spy(clear_spy).was.called(1)
+    end)
+
+    it("does not cancel a restore reclaimed after destroy", function()
+        local create_cb_ref = {}
+        local load_cb_ref = {}
+        local session = make_session(create_cb_ref, load_cb_ref)
+        local owner = session
```

Luacheck (1.2.0)

[warning] value assigned to variable 'owner' is overwritten on line 410 before use

(W311)